diff --git a/Test/Framework/Options.hs b/Test/Framework/Options.hs
index 30114dd..3f6177e 100644
--- a/Test/Framework/Options.hs
+++ b/Test/Framework/Options.hs
@@ -23,6 +23,9 @@ data TestOptions' f = TestOptions {
         -- ^ The number of microseconds to run tests for before considering them a failure
     }
 
+instance Semigroup (TestOptions' Maybe) where
+  (<>) = mappend
+
 instance Monoid (TestOptions' Maybe) where
     mempty = TestOptions {
             topt_seed = Nothing,
@@ -32,7 +35,7 @@ instance Monoid (TestOptions' Maybe) where
             topt_maximum_test_depth = Nothing,
             topt_timeout = Nothing
         }
-    
+
     mappend to1 to2 = TestOptions {
             topt_seed = getLast (mappendBy (Last . topt_seed) to1 to2),
             topt_maximum_generated_tests = getLast (mappendBy (Last . topt_maximum_generated_tests) to1 to2),
diff --git a/Test/Framework/Runners/Options.hs b/Test/Framework/Runners/Options.hs
index e6436b1..24ef10d 100644
--- a/Test/Framework/Runners/Options.hs
+++ b/Test/Framework/Runners/Options.hs
@@ -24,6 +24,9 @@ data RunnerOptions' f = RunnerOptions {
         ropt_list_only  :: f Bool
     }
 
+instance Semigroup (RunnerOptions' Maybe) where
+  (<>) = mappend
+
 instance Monoid (RunnerOptions' Maybe) where
     mempty = RunnerOptions {
             ropt_threads = Nothing,
diff --git a/Test/Framework/Runners/Statistics.hs b/Test/Framework/Runners/Statistics.hs
index ece6acb..902add0 100644
--- a/Test/Framework/Runners/Statistics.hs
+++ b/Test/Framework/Runners/Statistics.hs
@@ -30,6 +30,9 @@ adjustTestCount test_type amount = TestCount . Map.insertWith (+) test_type amou
 testCountTotal :: TestCount -> Int
 testCountTotal = sum . Map.elems . unTestCount
 
+instance Semigroup TestCount where
+  (<>) = mappend
+
 instance Monoid TestCount where
     mempty = TestCount $ Map.empty
     mappend (TestCount tcm1) (TestCount tcm2) = TestCount $ Map.unionWith (+) tcm1 tcm2
@@ -48,6 +51,9 @@ data TestStatistics = TestStatistics {
         ts_failed_tests :: TestCount
     }
 
+instance Semigroup TestStatistics where
+  (<>) = mappend
+
 instance Monoid TestStatistics where
     mempty = TestStatistics mempty mempty mempty mempty
     mappend (TestStatistics tot1 run1 pas1 fai1) (TestStatistics tot2 run2 pas2 fai2) = TestStatistics (tot1 `mappend` tot2) (run1 `mappend` run2) (pas1 `mappend` pas2) (fai1 `mappend` fai2)
